### PR TITLE
Add page with a delayed loading

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: src/main/kotlin/org/vividus/testsite/api/PageController.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Controller
+import org.springframework.ui.Model
+import org.springframework.ui.set
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestParam
+
+@Controller()
+class PageController @Autowired constructor(
+    private val resourceLoader: ResourceLoader
```
Comment (uarlouski) [inline src/main/kotlin/org/vividus/testsite/api/PageController.kt:16]: can be injected into getImageAsResponseEntity

Comment (valfirst) [reply src/main/kotlin/org/vividus/testsite/api/PageController.kt:16]: what's the profit?

File: src/main/kotlin/org/vividus/testsite/api/PageController.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        model["pageTimeout"] = pageTimeout
+        model["imageTimeout"] = imageTimeout
+        return "delayedLoading"
+    }
+
+    @GetMapping("/image")
+    fun getImageAsResponseEntity(@RequestParam(required = false, defaultValue = "0") timeout: Long): ResponseEntity<Any?> {
+        sleepFor(timeout)
+        val image: Resource = resourceLoader.getResource("classpath:vividus.png")
+        return ResponseEntity.ok(File(image.uri).readBytes())
```
Comment (valfirst) [inline src/main/kotlin/org/vividus/testsite/api/PageController.kt:34]: not each `Resource` can be converted to `File`, it's better to read `Resource` to bytes using `readableChannel()` or `getInputStream()`

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — CHANGES_REQUESTED** [review]
Also we already have the image: https://github.com/vividus-framework/vividus-test-site/blob/master/src/main/resources/static/img/vividus.png, it would be better to reuse it

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**